source/jormungandr/jormungandr/street_network/geovelo.py

```diff
@@ -354,27 +358,29 @@ def _direct_path(
         data = self._make_request_arguments_direct_path(
             pt_object_origin, pt_object_destination, request['bike_speed']
         )
+        single_result = True
+        if (
+            direct_path_type == StreetNetworkPathType.DIRECT
+            and request['direct_path'] == 'only_with_alternatives'
+        ):
+            single_result = False
         r = self._call_geovelo(
             '{}/{}'.format(
                 self.service_url,
                 'api/v2/computedroutes?'
                 'instructions=true&'
                 'elevations=true&'
                 'geometry=true&'
-                'single_result=true&'
+                'single_result={}&'
                 'bike_stations=false&'
-                'objects_as_ids=true&',
+                'objects_as_ids=true&'.format(single_result),
             ),
             requests.post,
             ujson.dumps(data),
         )
         self._check_response(r)
         resp_json = ujson.loads(r.text)
 
-        if len(resp_json) != 1:
```

Comment:
I'm not sure, but should we keep that (and then depend on if `single_result` is true or not for the test)?